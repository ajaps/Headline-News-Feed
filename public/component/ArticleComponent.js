import React from 'react';
import PropTypes from 'prop-types';

export default class ArticleComponent extends React.Component {

  render() {
    // Template for each Article
    return (
      <div className="col-lg-3 col-md-3 col-sm-3 col-xs-3">
        <a href={this.props.url} target="_Blank" rel="noopener noreferrer">
          <div className="thumbnail well text-center">
            <img src={this.props.urlToImage} alt="..." />
            <div className="caption">
              <p className="title">{this.props.title} </p>
              <p className="description">{this.props.description} </p>
              <p className="author">{this.props.author} </p>
              <p className="publishedAt">{this.props.publishedAt} </p>
            </div>
          </div>
        </a>
      </div>
    );
  }
}

ArticleComponent.propTypes = {
  url: PropTypes.string.isRequired,
  urlToImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
};
