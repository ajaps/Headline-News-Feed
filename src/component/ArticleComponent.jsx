import React from 'react';
import PropTypes from 'prop-types';

/**
 * Represents an Article.
 */
export default class ArticleComponent extends React.Component {

  /**
   * @returns {String} A function that retruns a custom date
   */
  formatDate() {
    const month = ['January', 'February', 'March',
      'April', 'May', 'June', 'July', 'August', 'September',
      'October', 'November', 'December'];

    const userDate = new Date(this.props.publishedAt);
    const cDate = userDate.getDate();
    const cMonth = userDate.getMonth();
    const cYear = userDate.getFullYear();

    const cHour = userDate.getHours();
    const cMin = userDate.getMinutes();
    const cSec = userDate.getSeconds();

    const formattedDate = `${month[cMonth]} ${cDate},   ${cYear}   ${cHour}:${cMin}:${cSec}`;
    return formattedDate;
  }

/**
 * @returns {component} A component with relevant article details.
 */
  render() {
    const getFormateDate = this.formatDate();
    return (
      <div className="col-lg-3 col-md-3 col-sm-3 col-xs-3">
        <a href={this.props.url} target="_Blank" rel="noopener noreferrer">
          <div className="thumbnail well text-center">
            <img src={this.props.urlToImage} alt="..." />
            <div className="caption">
              <p className="title">{this.props.title} </p>
              <p className="description">{this.props.description} </p>
              <p className="author">{this.props.author} </p>
              <p className="publishedAt">{getFormateDate} </p>
            </div>
          </div>
        </a>
      </div>
    );
  }
}


ArticleComponent.propTypes = {
  url: PropTypes.string.isRequired,
  urlToImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  author: PropTypes.string,
  publishedAt: PropTypes.string,
};

ArticleComponent.defaultProps = {
  publishedAt: 'unknown',
  author: 'unknown',
};
