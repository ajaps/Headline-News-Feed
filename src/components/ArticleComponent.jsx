import dateFormat from 'dateformat';
import PropTypes from 'prop-types';
import React from 'react';
import ReactLoading from 'react-loading';
import TextTruncate from 'react-text-truncate';

/**
 * Represents an Article.
 */
export default class ArticleComponent extends React.Component {

/**
 * @returns {component} A component with relevant article details.
 */
  render() {
    const error = this.props.error;
    const articles = this.props.articles;

    const articleComponents = articles.map((articleItem) => {
    // change custom date to a more readable date
      const articlePublishDate = new Date(articleItem.publishedAt);
      const getFormateDate = dateFormat(articlePublishDate);
      return (
        <div key={articleItem.url}>
          <div className="col-lg-3 col-md-3 col-sm-3 col-xs-3">
            <a href={articleItem.url} target="_Blank" rel="noopener noreferrer">
              <div className="thumbnail well text-center">
                <img src={articleItem.urlToImage} alt="Not Available" />
                <div className="caption">
                  <p className="title">{articleItem.title} </p>
                  <div className="article-description ">
                    <TextTruncate
                    line={2}
                    truncateText="....."
                    text={articleItem.description}
                    />
                  </div>
                  <p className="publishedAt">{getFormateDate} </p>
                </div>
              </div>
            </a>
          </div>
        </div>
      );
    });

    return error ?
      <div className="container well outerArticleCountainer">
        <ReactLoading className="spin" width="99px"
        type="spin" color="#3b5998"/>
      </div> :
    (
      <div className="container well outerArticleCountainer">
        <div className="row article-Container">{articleComponents} </div>
      </div>
    );
  }
}


ArticleComponent.propTypes = {
  articles: PropTypes.array.isRequired,
  error: PropTypes.string,
};
