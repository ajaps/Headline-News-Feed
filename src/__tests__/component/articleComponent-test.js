import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import ArticleComponent from '../../components/ArticleComponent.jsx';

describe('Category', () => {
  let app;
  const mockArticle =
    [
      {
        author: 'Ajaps Franklin',
        title: 'Nigeria has emerged as a developed country',
        description: 'Nigeria is now one og the most developed country in the',
        url: 'https://thenextweb.com/insider/2017/06/14/pornhub-piss-off-entir',
        urlToImage: 'https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files.jpg',
        publishedAt: '2018-06-06T21:20:41Z'
      },
      {
        author: 'tnude Abiola',
        title: 'A Nigerian manufactured the fastest car',
        description: 'A nigerian by the name of obinna manufactured the',
        url: 'https://bbc-news.com/insider/2017/06/14/pornhub-piss-off-entir',
        urlToImage: 'https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files.jpg',
        publishedAt: '2018-07-06T21:20:41Z'
      },
    ];

  beforeEach(() => {
    app = shallow(<ArticleComponent articles={mockArticle} error={'error'}/>);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should have extacly two div element if an Error occured', () => {
    expect(app.find('div').length).toEqual(2);
  });

  it('will contain the same number of TextTruncate as the array passed', () => {
    const appNoError = shallow(<ArticleComponent articles={mockArticle} />);
    expect(appNoError.find('TextTruncate').length).toEqual(2);
  });
});
