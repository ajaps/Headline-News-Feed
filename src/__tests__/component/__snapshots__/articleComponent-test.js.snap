// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should render as expected 1`] = `
<div
  className="container well outerArticleCountainer"
>
  <div
    className="row article-Container"
  >
    <Error
      error="error"
    />
     
  </div>
</div>
`;
